### ISSUE TITLE
Disabling reconnection I do not know when it was disconnected.

### ISSUE THREAD

**Issue by rudacs** [mqttjs/MQTT.js#704]
Disabling reconnection I do not know when it was disconnected.
The disconnect event only works if reconnection is enabled. What does not make sense.

**Comment by mcollina** [comment]
Can you rephrase? I'm not understanding what problem you are having.
Can you show what behavior you would like to have with a code example?

**Comment by noerw** [comment]
@rudacs I guess you are talking about the `reconnect` event, there is no `disconnect` event afaik. You can use the `offline` event to get notified about a broken connection. It is only emitted after the keepalive has timed out (which was after 120 seconds for me).

Subscribing to `connect`, `reconnect`, `close`, `offline` events leads to the following flow for me (with autoreconnect enabled):
```
20:01:45.640Z mqtt connect
20:01:50.226Z      wifi disabled
20:03:45.647Z mqtt offline
20:03:45.647Z mqtt close
20:03:46.648Z mqtt reconnect
20:03:46.648Z mqtt close
20:03:47.648Z mqtt reconnect
20:03:47.648Z mqtt close
20:03:47.926Z       wifi enabled
20:03:48.648Z mqtt reconnect
20:03:48.648Z mqtt close
20:03:49.648Z mqtt reconnect
20:03:49.648Z mqtt close
20:03:50.648Z mqtt reconnect
20:03:50.982Z mqtt connect
```
